utils/cacharr_settings.py

```
@@ -0,0 +1,95 @@
+    instances = prowlarr_cfg.get("instances") or {}
+    if not isinstance(instances, dict):
+        logger.warning("Cacharr: invalid prowlarr.instances type: %s", type(instances).__name__)
+        return "", ""
+    for _inst_key, inst in instances.items():
+        if not isinstance(inst, dict) or not inst.get("enabled"):
+            continue
+        config_file = (inst.get("config_file") or "").strip()
+        api_key = ""
+        if config_file and os.path.isfile(config_file):
+            try:
+                tree = ET.parse(config_file)
```

Ruff (0.15.6)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)